### ISSUE TITLE
Add ruby 2.3.3 dependency for bootstrap-sass 3.4.0 

### ISSUE THREAD

**Issue by vid-pas** [twbs/bootstrap-sass#1182]
Is there a place to note Ruby dependencies for bootstrap-sass? 
It appears that bootstrap-sass 3.4.0 requires Ruby 2.3.3.

----

This is a relevant problem when installing ruby and bootstrap-sass via `apt'. 

By default apt installs the latest bootstrap-sass (currently 3.4.0) and ruby 2.3.1.
Here's the error I get when building a server with Ansible:
`ERROR:  Error installing bootstrap-sass:\n\tsassc requires Ruby version >= 2.3.3.`

Apt is limited to Ruby 2.3.1; Per: https://www.ruby-lang.org/en/documentation/installation/   "As of writing, the ruby-full package provides Ruby 2.3.1, which is an old stable release, on Debian and Ubuntu."

It would be ideal if this could be updated in the same place that other dependencies are set for ruby gems. When I look here for example: https://rubygems.org/gems/bootstrap-sass/versions/3.4.0
It reads: "Required Ruby Version: >= 0"

Maybe in bootstrap-sass.gemspec?

**Comment by komlenic** [comment]
I ran into this issue as well when working with bootstrap-sass 3.4 in Drupal 8 and seeing 'bundle install' failing. Ruby 2.0.x is the latest version available to install on CentOS via yum and regular repositories. Once I installed Ruby 2.3.3 via RVM, everything works correctly, but I only arrived at that solution by discovering this GitHub issue.

**Comment by DeeDeeG** [comment]
I found a way to specify required ruby versions (as you had guessed, it is indeed via the .gemspec file): https://stackoverflow.com/questions/5004492/how-do-you-specify-a-minimum-ruby-version-in-a-gemspec

```ruby
spec.required_ruby_version ='>= 2.3.3'
```
or
```ruby
spec.required_ruby_version ='~> 2.3.3'
```
etc.

**Comment by glebm** [comment]
I suspect sassc doesn't really need 2.3.3. Consider checking if it works with 2.3.1 and sending a PR to sassc

**Comment by DeeDeeG** [comment]
I filed https://github.com/sass/sassc-ruby/issues/106 to see if `sassc-ruby` would be interested in lowering their required Ruby version number.

**Comment by DeeDeeG** [comment]
@vid-pas, @komlenic,

If you folks need a workaround, you can:

- Change the line `spec.required_ruby_version = [version]` of the `sassc.gemspec` filein the sassc repo like so:

```diff
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})

-  spec.required_ruby_version = ">= 2.3.3"
+  spec.required_ruby_version = ">= 2.0.0"

   spec.require_paths = ["lib"]
```

- And specify this patched sassc in your Gemfile like so:

```ruby
gem 'sassc', git: 'https://github.com/[username]/sassc-ruby', ref: '[commit-hash]'
```
example:
```ruby
gem 'sassc', git: 'https://github.com/DeeDeeG/sassc-ruby', ref: '19f07dfc2790173f8a73fe0bfa1310fcdbbd63ce'
```

(You can use [my fork](https://github.com/DeeDeeG/sassc-ruby/tree/Ruby-2.0.0), based off of `sassc-ruby`'s master branch as of a few days ago... But I'm not planning to update or maintain the fork, so it will be out-of-date very quickly. I recommend either manually patching as I described (so you can update and re-patch as needed), or politely making some noise about this over at the [sass/sassc-ruby](https://github.com/sass/sassc-ruby/issues/106) repo.)

Best,

\- DeeDeeG

**Comment by dracmaAgil** [comment]
I'm using rbenv and update ruby-build: cd /home/dracma/.rbenv/plugins/ruby-build && git pull && cd -
and then list again ruby versions and there are updated, 2.3.2-2.3.8. I hope this works for you guys!

**Comment by DeeDeeG** [comment]
Yeah, using a ruby version manager, like [RVM](https://github.com/rvm/rvm) or [rbenv](https://github.com/rbenv/rbenv) is the main way to get ruby outside of the ubuntu/fedora/centos(etc...) repositories.

(I prefer rbenv, personally. And with rbenv, it's good to keep rbenv and the [ruby-build](https://github.com/rbenv/ruby-build) plugin up-to-date, as @dracmaAgil says.)

**Comment by Huixh111** [comment]
o
